packages/catalog-process/src/services/catalogService.ts

```diff
@@ -326,6 +326,15 @@ async function parseAndCheckAttributes(
   };
 }
 
+function isTenantInSignalHubWhitelist(
+  organizationId: TenantId,
+  isSignalubEnabled: boolean | undefined
+): boolean | undefined {
+  return config.signalhubWhitelist?.includes(organizationId)
+    ? isSignalubEnabled
+    : false;
```

Comment:
```suggestion
): boolean | undefined {
  return (
    config.featureFlagSignalhubWhitelist &&
    config.signalhubWhitelist?.includes(organizationId)
  );
```
I suggest to implement it like this, in order to not be dependant on `isSignalubEnabled`, see my other suggestions on where we use it.

Moreover, I would move this function in `validators`

Reply:
What happens if `FEATURE_FLAG_SIGNALHUB_WHITELIST` is false? that's variable set to false means that you should not consider whitelist and `isSignalubEnabled` has to be set inheriting value comes from UI. In this case you wouldn't consider `isSignalubEnabled` in any case.

Reply:
> What happens if `FEATURE_FLAG_SIGNALHUB_WHITELIST` is false? that's variable set to false means that you should not consider whitelist and `isSignalubEnabled` has to be set inheriting value comes from UI. In this case you wouldn't consider `isSignalubEnabled` in any case.

You're right! You can discard these suggestions